### ISSUE TITLE
[BUG] Cluster Member cannot see clusternetworks

### ISSUE THREAD

**Issue by futuretea** [harvester/harvester#3300]
**Describe the bug**

**To Reproduce**
1. Imported Harvester into Rancher under Virtualization Management.
2. Created a new  standard user into Rancher.
3.  Under Virtualization Management, edit the Harvester configuration and add the new users as Cluster Members.
![image](https://user-images.githubusercontent.com/15064560/208407408-164f3d53-06fe-4fcf-8a9e-bb4e30ccf8b1.png)
![image](https://user-images.githubusercontent.com/15064560/208407429-8c63de83-9534-41fd-9efe-937727875ab3.png)
4. Using the new member users, log into the Rancher platform and open the Harvester management page.
5. Under Networks >> VM Networks >> Create, there is a field called ¨Cluster Network¨. When the user opens the dropdown, there is no network available, and the message ¨Sorry, no matching options¨
![image](https://user-images.githubusercontent.com/15064560/208407669-b84904dd-6a2e-42e4-ae0c-26c00a91be32.png)

**Expected behavior**

Expect to see and select available cluster networks.

**Support bundle**

**Environment**
 - Harvester ISO version: v1.1
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**

**Comment by futuretea** [comment]
Workaround: apply the patched ClusterRole harvester-authenticated after host reboot or harvester-system/harvester pod restart 

```bash
kubectl apply -f ./harvester-authenticated.yaml
```

The content of harvester-authenticated.yaml
```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: harvester-authenticated
rules:
- apiGroups:
  - harvesterhci.io
  resources:
  - settings
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - network.harvesterhci.io
  resources:
  - clusternetworks
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - ""
  resourceNames:
  - harvester-public
  resources:
  - namespaces
  verbs:
  - get
  - watch
```


and change `verbs` of network-attachment-definitions to `"*"` by

```
kubectl edit clusterRole harvesterhci.io:edit
```

**Comment by TachunLin** [comment]
Verified **fixed** on `master-14bbee2c-head` (1/13) with Rancher v2.7.0. Close this issue.

### Result
Create a normal user with cluster member role to access Harvester from virtualization management. 
The user can see and select the clusternetworks item


https://user-images.githubusercontent.com/29251855/213165501-95c7d6c2-5f97-4ffe-80db-74d6671939cc.mp4



### Test Information
* Test Environment: 1 nodes harvester on local kvm machine
* Harvester version: `master-14bbee2c-head` (1/13)
* Rancher version: v2.7.0

### Verify Steps
Steps to reproduce the behavior:
1. Imported Harvester into Rancher under Virtualization Management.
2. Created a new  standard user into Rancher.
3.  Under Virtualization Management, edit the Harvester configuration and add the new users as Cluster Members.
4. Using the new member users, log into the Rancher platform and open the Harvester management page.
5. Under Networks >> VM Networks >> Create, there is a field called ¨Cluster Network¨. When the user opens the dropdown, there is no network available, and the message ¨Sorry, no matching options¨
